okio-behavior/src/main/java/okio/test/BehaviorSink.java

```diff
@@ -0,0 +1,62 @@
+package okio.test;
+
+import java.io.IOException;
+import okio.Buffer;
+import okio.ForwardingSink;
+import okio.Sink;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+/** A wrapper which applies a {@link Behavior} to a {@link Sink}. */
+public final class BehaviorSink extends ForwardingSink {
+  private final Behavior behavior = new Behavior();
+
+  public BehaviorSink(Sink delegate) {
+    super(delegate);
+  }
+
+  public Behavior behavior() {
+    return behavior;
+  }
+
+  @Override public void write(Buffer source, long byteCount) throws IOException {
```

Comment:
How do you mean? No matter how many of how few calls you make we chunk them and sleep accordingly. And if you make many calls we adjust the sleep based on the partial amount.